### PR TITLE
Refs #21012 -- Removed unnecessary _create_cache() hook.

### PR DESCRIPTION
**Pull request by felixxm**
This removed unused workaround to load a cache backend with its dotted import path.

Unused since d038c547b5ce585cbf9ef5bb7e5298f52e4a243b.

### CODE REVIEW

File: django/core/cache/__init__.py [hunk 1]
```diff
@@ -31,22 +31,9 @@
...
-            except ImportError as e:
-                raise InvalidCacheBackendError("Could not find backend '%s': %s" % (
-                    backend, e))
-            location = kwargs.pop('LOCATION', '')
-            params = kwargs
-        else:
-            params = {**conf, **kwargs}
-            backend = params.pop('BACKEND')
-            location = params.pop('LOCATION', '')
+        params = {**settings.CACHES[backend], **kwargs}
```
Comment (timgraham) [inline django/core/cache/__init__.py:34]: Can this fail with KeyError? It looks like that was transformed to InvalidCacheBackendError in the old code.

File: django/core/cache/__init__.py [hunk 2]
```diff
@@ -31,22 +31,9 @@
...
+        backend = params.pop('BACKEND')
+        location = params.pop('LOCATION', '')
         backend_cls = import_string(backend)
```
Comment (timgraham) [inline django/core/cache/__init__.py:37]: It looks like only this line needs to be in the try/except?

File: django/core/cache/__init__.py (hunk 1 above)
Comment (felixxm) [reply django/core/cache/__init__.py:34]: It's used only in `CacheHandler` where we check that alias is correct: https://github.com/django/django/blob/7ec2658e1e24149f0f3244c08c361348f6ebc0e4/django/core/cache/__init__.py#L74-L77

Comment (timgraham) [reply django/core/cache/__init__.py:34]: Maybe the separate method could be removed?

Comment (felixxm) [reply django/core/cache/__init__.py:34]: Sounds good, I will prepare edits later today.

**Comment by felixxm** [comment]
@timgraham I pushed edits, thanks for the review :+1:

**Review by timgraham — APPROVED** [review]

**Pull request merged**